src/main/java/com/connexta/store/controllers/StoreController.java

```diff
@@ -129,6 +109,72 @@
     return ok().build();
   }
 
+  @Override
+  public ResponseEntity<Void> ingest(
+      String acceptVersion,
+      @DateTimeFormat(iso = DateTimeFormat.ISO.DATE_TIME) OffsetDateTime lastModified,
+      MultipartFile file,
+      String correlationId,
+      MultipartFile metacard) {
+    if (lastModified == null || lastModified.toString().isBlank()) {
+      throw new ServerWebInputException(
+          String.format("%s is missing or blank", LAST_MODIFIED)); // TODO Replace this exception
+    }
+    String fileName = file.getOriginalFilename();
+    log.info("Ingest request received fileName={}", fileName);
+    InputStream inputStream;
+    InputStream metacardInputStream;
+    try {
+      inputStream = file.getInputStream();
+      metacardInputStream = metacard.getInputStream();
+    } catch (IOException e) {
+      throw new ValidationException("Could not open attachment"); // TODO Replace this exception
+    }
+    storeService.ingest(
+        file.getSize(),
+        file.getContentType(),
+        inputStream,
+        fileName,
+        metacard.getSize(),
+        metacardInputStream);
+
+    return ResponseEntity.accepted().build();
+  }
+
+  @GetMapping("/{id}")
```

Comment:
Should we update this path? E.g. `/tmp/{metacardId}` or `/dataset/{datasetId}/metacard`?

Reply:
added `public static final String RETRIEVE_METACARD_URL_TEMPLATE = "/dataset/{datasetId}/metacard";`